index.js

```diff
@@ -1,110 +1,51 @@
-// ***************************************************************************
-// Iteration 1 - `for...of` loop
-// ***************************************************************************
 
+// Iteration 1 
 const getFirstNames = arr => {
   const userFirstNames = [];
   for (let user of arr) {
-    // Your code goes here ...
+    userFirstNames.push(user.firstName)
   }
+    return userFirstNames
 };
-
 getFirstNames(usersArray);
-// expected output:
-// [ 'Kirby', 'Tracie', 'Kendra', 'Kinney', 'Howard', 'Rachelle', 'Lizzie' ]
-
-// ***************************************************************************
-// Iteration 2 - `for...of` loop and ES6 string literals `${}`
-// ***************************************************************************
 
+// Iteration 2 
 const getFullNames = arr => {
-  // Your code goes here ...
+  const userfullNames = [];
+for(let user of arr) {
+  userfullNames.push(`${user.firstName} ${user.lastName}`)
+}
+return userfullNames
 };
-
 getFullNames(usersArray);
-// expected output:
-// [ 'Kirby Doyle', 'Tracie May', 'Kendra Hines', 'Kinney Howard',
-//   'Howard Gilmore', 'Rachelle Schneider', 'Lizzie Alford' ]
-
-// ***************************************************************************
-// Iteration 3 - ES6 destructuring , for of loop, object literal
-// ***************************************************************************
 
+// Iteration 3 
 const getUsersCreditDetails = arr => {
-  // Your code goes here ...
-};
-
+  const usersCreditDetails = [];
+  for(let {firstName, lastName, balance} of arr){
+    const userDetails = {
+      firstName,
+      lastName,
+      balance
+    }
+    usersCreditDetails.push(userDetails)
```

Comment:
nice job in destructuring the variables directly in the for loop. this is a more advanced thing 👍 

you could have done userCreditDetails.push({firstName, lastName, balance}) directly